lib/ramble/ramble/spack_runner.py

```diff
@@ -425,16 +427,40 @@ def generate_env_file(self):
         """
         self._check_active()
 
-        env_file = {}
-        env_file[spack_namespace] = {}
-        env_file[spack_namespace]['concretizer'] = {}
+        env_file = syaml.syaml_dict()
+        env_file[spack_namespace] = syaml.syaml_dict()
+        env_file[spack_namespace]['concretizer'] = syaml.syaml_dict()
         env_file[spack_namespace]['concretizer']['unify'] = True
 
-        env_file[spack_namespace]['specs'] = []
+        env_file[spack_namespace]['specs'] = syaml.syaml_list()
         env_file[spack_namespace]['specs'].extend(self.env_contents)
 
         env_file[spack_namespace]['include'] = self.includes
 
+        spack_env_file = os.path.join(self.env_path, 'spack.yaml')
+        spack_lock_file = os.path.join(self.env_path, 'spack.lock')
+
+        # Check that a spack.yaml and spack.lock file exist already
+        if os.path.exists(spack_env_file) and os.path.exists(spack_lock_file):
+            existing_env_mtime = os.path.getmtime(spack_env_file)
+            existing_lock_mtime = os.path.getmtime(spack_lock_file)
+
+            # If the lock file was last modified after the yaml file...
+            if existing_lock_mtime >= existing_env_mtime:
```

Comment:
I was thinking about what it means for these times to be equal, and it seems unlikely that in normal use they would be equal? 

One interesting case where they might be equal is if someone copy paste the dir structure from somewhere (I think?). In that case it might be worth regenerating 

Are there other reasonable reasons these would be equal?

Reply:
Good point. I can't decide how I feel about the copy/paste. But I think reconcretizing wouldn't hurt in that case.